cmd/snap-bootstrap/cmd_initramfs_mounts.go

```diff
@@ -178,6 +178,44 @@ func generateInitramfsMounts() (err error) {
 	return nil
 }
 
+func doBaseBindMount() (err error) {
+	// Bind-mount base snap to /sysroot as the start of pivot-root
+	// rootfs. Previously this was in the core-initrd, but it is
+	// best to be controlled by snap-bootstrap. NB! This needs to
+	// be backwards compatible, because older initrds are often
+	// repacked with newer snap-bootstrap.
+	bindSystemdOpts := &systemdMountOptions{
+		Bind: true,
+	}
+	if !osutil.FileExists(filepath.Join(dirs.GlobalRootDir, "/usr/lib/systemd/system/sysroot.mount")) {
```

Comment:
nice, we had it on our long todo list to do this, I like this method of doing it